### PR TITLE
Support iterate a dataset in sequential order when training

### PR DESCRIPTION
**Pull request by yangw1234**
It is more common to iterate a dataset in sequential order than to random sample with replacement.

### CODE REVIEW

**Comment by qiuxin2012** [comment]
Please add some unit test,

**Comment by yangw1234** [comment]
> Please add some unit test,

Sure, working on it

**Comment by yangw1234** [comment]
unit tests added

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2727/

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/FeatureSet.scala [hunk 1]
```diff
@@ -230,31 +232,52 @@ class CachedDistributedFeatureSet[T: ClassTag]
...
 
-  protected var indexes: RDD[Array[Int]] = buffer.mapPartitions(iter => {
-    Iterator.single[Array[Int]]((0 until iter.next().length).toArray[Int])
+  protected var indexes: RDD[(Array[Int], AtomicInteger)] = buffer.mapPartitions(iter => {
+    Iterator.single[(Array[Int], AtomicInteger)](((0 until iter.next().length).toArray[Int],
+      new AtomicInteger(0)))
   }).setName(s"origin index of ${buffer.name}").cache()
 
 
+  protected var offsets: RDD[AtomicInteger] = buffer.mapPartitions(iter => {
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/feature/FeatureSet.scala]: looks like useless?

Comment (yangw1234) [reply zoo/src/main/scala/com/intel/analytics/zoo/feature/FeatureSet.scala]: good catch! I'll delete it.

**Pull request merged**